### ISSUE TITLE
Momentum and learning rate annealing for gradient descent

### ISSUE THREAD

**Issue by axiomizer** [axiomizer/haida-go#3]
![image](https://github.com/axiomizer/haida-go/assets/138620999/e1c59694-7dc2-43f7-ac4d-824668b587c4)

**Comment by axiomizer** [comment]
learning rate annealing (with unit test) implemented in 270b8c424a60984d6e9ef706f50c4efe79cbd085
